Tests/SonarQube.TeamBuild.PreProcessor.Tests/TargetsInstallerTests.cs

```diff
@@ -75,22 +87,245 @@ public void InstallTargetsFile_Overwrite()
+                sourceContent: "target content",
+                destinationExists: true,
+                destinationContent: "target content");
+
+            Assert.IsTrue(logger.DebugMessages.Any(m =>
+                m.Equals("The file SonarQube.Integration.targets is up to date at c:\\project\\bin\\targets")));
+        }
+
+        [TestMethod]
+        public void InstallLoaderTargets_InternalCopyTargetFileToProject_Different_Content()
+        {
+            InstallLoaderTargets_InternalCopyTargetFileToProject(
+                sourceContent: "target content",
+                destinationExists: true,
+                destinationContent: "different content");
+
+            Assert.IsTrue(logger.DebugMessages.Any(m =>
+                m.Equals("The file SonarQube.Integration.targets was overwritten at c:\\project\\bin\\targets")));
+        }
+
+        [TestMethod]
+        public void InstallLoaderTargets_InternalCopyTargetFileToProject_Not_Exists()
+        {
+            InstallLoaderTargets_InternalCopyTargetFileToProject(
+                sourceContent: "target content",
+                destinationExists: false);
+
+            Assert.IsTrue(logger.DebugMessages.Any(m =>
+                m.Equals("Installed SonarQube.Integration.targets to c:\\project\\bin\\targets")));
+        }
+
+        [TestMethod]
+        public void InstallLoaderTargets_InternalCopyTargetsFile_Same_Content()
+        {
+            InstallLoaderTargets_InternalCopyTargetsFile(
+                sourceContent: "target content",
+                destinationExists: true,
+                destinationContent: "target content");
+
+            Assert.IsTrue(logger.DebugMessages.Any(m =>
+                m.Equals("The file SonarQube.Integration.ImportBefore.targets is up to date at c:\\global paths")));
+        }
+
+        [TestMethod]
+        public void InstallLoaderTargets_InternalCopyTargetsFile_Different_Content()
+        {
+            InstallLoaderTargets_InternalCopyTargetsFile(
+                sourceContent: "target content",
+                destinationExists: true,
+                destinationContent: "different content");
+
+            Assert.IsTrue(logger.DebugMessages.Any(m =>
+                m.Equals("The file SonarQube.Integration.ImportBefore.targets was overwritten at c:\\global paths")));
+        }
+
+        [TestMethod]
+        public void InstallLoaderTargets_InternalCopyTargetsFile_Not_Exists()
+        {
+            InstallLoaderTargets_InternalCopyTargetsFile(
+                sourceContent: "target content",
+                destinationExists: false);
+
+            Assert.IsTrue(logger.DebugMessages.Any(m =>
+                m.Equals("Installed SonarQube.Integration.ImportBefore.targets to c:\\global paths")));
+        }
+
+        private void InstallLoaderTargets_InternalCopyTargetFileToProject(string sourceContent, bool destinationExists, string destinationContent = null)
```

Comment:
This is mostly duplicate code with the next method, but too many arguments will make the tests harder to understand.